tests/util/simple_tests.rs

```diff
@@ -25,22 +25,50 @@ use self::tempdir::TempDir;
 use self::uuid::Uuid;
 
 use libstratis::engine::{Engine, EngineError, ErrorEnum};
+use libstratis::engine::types::{FilesystemUuid, Redundancy};
 use libstratis::engine::strat_engine::blockdevmgr::{initialize, resolve_devices};
 use libstratis::engine::strat_engine::device::{blkdev_size, wipe_sectors, write_sectors};
 use libstratis::engine::strat_engine::engine::DevOwnership;
 use libstratis::engine::strat_engine::filesystem::{create_fs, mount_fs, unmount_fs};
 use libstratis::engine::strat_engine::metadata::{StaticHeader, BDA_STATIC_HDR_SECTORS,
                                                  MIN_MDA_SECTORS};
-use libstratis::engine::strat_engine::pool::{get_dmdevs, get_filesystems};
+use libstratis::engine::strat_engine::pool::{get_dmdevs, get_filesystems, DATA_BLOCK_SIZE,
+                                             INITIAL_DATA_SIZE, StratPool};
 use libstratis::engine::strat_engine::serde_structs::Recordable;
 use libstratis::engine::strat_engine::setup::{find_all, get_blockdevs, get_metadata};
 use libstratis::engine::strat_engine::StratEngine;
+use libstratis::engine::Pool;
 
 /// Dirty sectors where specified, with 1s.
 fn dirty_sectors(path: &Path, offset: Sectors, length: Sectors) {
     write_sectors(path, offset, length, &[1u8; SECTOR_SIZE]).unwrap();
 }
 
+/// Verify that a the physical space allocated to a pool is expanded when
+/// the nuber of sectors written to a thin-dev in the pool exceeds the
+/// INITIAL_DATA_SIZE.  If we are able to write more sectors to the filesytem
+/// than are initially allocated to the pool, the pool must have been expanded.
+fn verify_pool_expand(pool: &mut StratPool, fs_uuid: &FilesystemUuid) {
+    let devnode = pool.get_filesystem(&fs_uuid)
+        .unwrap()
+        .devnode()
+        .unwrap();
+    let mut f = OpenOptions::new().write(true).open(devnode).unwrap();
+    // Write 10 percent more sectors than is initially allocated to a pool
+    let write_size: u64 = (INITIAL_DATA_SIZE.0 as f64 +
+                           (INITIAL_DATA_SIZE.0 as f64 * (0.10 as f64 / 100.0 as f64)))
+            .round() as u64;
+    let buf = &[1u8; SECTOR_SIZE];
+    for i in 0..write_size {
+        f.write_all(buf).is_ok();
+        // Run a pool.check() for each block written.  This simulates an event that would be
+        // handled by Stratis.
+        if i % DATA_BLOCK_SIZE.0 == 0 {
```

Comment:
We need to implement something that listens to DM events.  When a "low water mark crossed" event is triggered - Stratis will field the event and call check on the pool.  

Until we've got a listener - I'm manually calling check in the test.  How about if I open an issue, add a TODO to the code?

I could also change the code to only call the check() function when we are within X (say 100) sectors of the INITIAL_DATA_SIZE?